bin/wp-cli.php

```diff
@@ -276,35 +298,93 @@ public function index( $args, $assoc_args ) {
 		$this->finish( $timestamp_start );
 	}
 
-
 	/**
-	 * Index a single post in elasticsearch and output debugging information. You should enable SP_DEBUG and SAVEQUERIES before running this.
+	 * Index a single post in ElasticSearch and output debugging information. You should enable `SP_DEBUG` and `SAVEQUERIES` before running this.
+	 *
+	 * <post_id>
+	 * : Post ID to index.
 	 *
-	 * @synopsis <post_id>
+	 * ## EXAMPLE
+	 *
+	 *     $ wp searchpress debug 123
+	 *     Index complete!
 	 */
 	public function debug( $args ) {
-		if ( empty( $args[0] ) )
-			WP_CLI::error( "Invalid post ID" );
+		global $wpdb;
+
 		$post_id = intval( $args[0] );
-		if ( ! $post_id )
-			WP_CLI::error( "Invalid post ID" );
 
-		global $wpdb;
+		if ( empty( $post_id ) ) {
+			WP_CLI::error( 'Invalid post ID.' );
+		}
+
+		if ( ! defined( 'SP_DEBUG') ) {
+			WP_CLI::error( 'SP_DEBUG should be enabled before running this.' );
+		}
+
+		if ( ! defined( 'SAVEQUERIES') ) {
+			WP_CLI::error( 'SAVEQUERIES should be enabled before running this.' );
+		}
```

Comment:
ooc, did you happen to check if it's too late at this point to set those? would be great to not require it.

Reply:
at the very least, I think `error()` is too strong since it will kill the script. a warning is good enough.

Reply:
Cool, will update! I interpreted _should_ as a _must_.